### PR TITLE
Feature/map

### PR DESCRIPTION
**Pull request by Swess**
A not 100% perfect Graph with Continent Subgraphs.

### CODE REVIEW

File: src/Map/Map.cpp [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+#include <iostream>
+#include <list>
+
+namespace Map {
+
+    Country::Country(const string &name) {
+        this->name = new string(name);
+        this->index = new int(-1);
+        this->continent_index = new int(-1);
+        this->nb_armies = new int(0);
```
Comment (Kerry-G) [inline src/Map/Map.cpp:11]: what's the difference between init -1 or 0

File: src/Map/MapDriver.h [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#pragma once
+
+namespace Map {
+    namespace Driver {
```
Comment (Kerry-G) [inline src/Map/MapDriver.h:4]: just wondering why namespace over classes?

**Review by Kerry-G — APPROVED** [review]
just had two questions about implementation design. If there's no reasons for the difference between -1 and 0 go ahead and merge, otherwise just fix that and it's fine

File: src/Map/Map.cpp (hunk 1 above)
Comment (Swess) [reply src/Map/Map.cpp:11]: 0 would mean index 0... -1 means no index set yet

File: src/Map/MapDriver.h (hunk 2 above)
Comment (Swess) [reply src/Map/MapDriver.h:4]: Euhm.. no object creation necessary that way. Can be changed if ever.

**Pull request merged**